WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListFragment.java

```diff
@@ -1204,74 +1184,117 @@ private void setEmptyTitleAndDescription(boolean requestFailed) {
+    private void setEmptyTitleDescriptionAndButton(@NonNull String title, String description,
+                                                   final ActionableEmptyViewButtonType button) {
         if (!isAdded()) {
             return;
         }
 
-        TextView titleView = mEmptyView.findViewById(R.id.title_empty);
-        titleView.setText(title);
+        mActionableEmptyView.setImageVisibility(!isUpdating());
+        mActionableEmptyView.setTitleText(title);
 
-        TextView descriptionView = mEmptyView.findViewById(R.id.description_empty);
         if (description == null) {
-            descriptionView.setVisibility(View.INVISIBLE);
+            mActionableEmptyView.setSubtitleVisibility(false);
         } else {
+            mActionableEmptyView.setSubtitleVisibility(true);
+
             if (description.contains("<") && description.contains(">")) {
-                descriptionView.setText(Html.fromHtml(description));
+                mActionableEmptyView.setSubtitleText(Html.fromHtml(description));
             } else {
-                descriptionView.setText(description);
+                mActionableEmptyView.setSubtitleText(description);
             }
-            descriptionView.setVisibility(View.VISIBLE);
         }
 
-        mEmptyViewBoxImages.setVisibility(shouldShowBoxAndPagesAnimation() ? View.VISIBLE : View.GONE);
+        if (button == null) {
+            mActionableEmptyView.setButtonVisibility(false);
+        } else {
+            mActionableEmptyView.setButtonVisibility(true);
+
+            switch (button) {
+                case DISCOVER:
+                    mActionableEmptyView.setButtonText(getString(R.string.reader_empty_followed_blogs_button_discover));
+                    break;
+                case FOLLOWED:
+                    mActionableEmptyView.setButtonText(getString(R.string.reader_empty_followed_blogs_button_followed));
+                    break;
+            }
+
+            mActionableEmptyView.setButtonClickListener(new View.OnClickListener() {
+                @Override public void onClick(View view) {
+                    setCurrentTagFromEmptyViewButton(button);
+                }
+            });
+        }
     }
 
     private void showEmptyView() {
         if (isAdded()) {
-            mEmptyView.setVisibility(View.VISIBLE);
+            mActionableEmptyView.setVisibility(View.VISIBLE);
         }
     }
 
     private void hideEmptyView() {
         if (isAdded()) {
-            mEmptyView.setVisibility(View.GONE);
+            mActionableEmptyView.setVisibility(View.GONE);
         }
     }
 
     private boolean isEmptyViewShowing() {
-        return isAdded() && mEmptyView.getVisibility() == View.VISIBLE;
+        return isAdded() && mActionableEmptyView.getVisibility() == View.VISIBLE;
+    }
+
+    private void setCurrentTagFromEmptyViewButton(ActionableEmptyViewButtonType button) {
+        ReaderTag tag = ReaderUtils.getDefaultTag();
```

Comment:
Just a tiny nitpick - maybe we could put this into a default case of switch statement?

Reply:
I updated the tag assignment in 9453f23.